src/LanguageServer.ts

```diff
@@ -903,8 +903,8 @@ export class LanguageServer {
 
             await this.sendDiagnostics();
         } catch (e) {
-            this.connection.tracer.log(e);
-            this.sendCriticalFailure(`Critical error validating workspace: ${e.message}`);
+            this.connection.console.error(e);
```

Comment:
The tracer log wasn't showing up in the output panel, but `this.connection.console.error` does. Not relevant to the fix, but useful to have overall.

Reply:
> Nice one... Parsers with recovery can really produce wonky AST 😬

Yeah definitely. I think I introduced this bug when I added the for loop item to completions.